### PR TITLE
Added some unit tests to addrmgr.

### PR DESCRIPTION
**Pull request by jimmysong**
Unit tests to get a little more coverage into the addrmgr.

### CODE REVIEW

**Comment by davecgh** [comment]
I haven't reviewed this yet, but very timely!  I was just talking to @dajohi yesterday evening about getting the coverage up on this package.

File: addrmgr/addrmanager_test.go [hunk 1]
```diff
@@ -102,35 +107,99 @@ func lookupFunc(host string) ([]net.IP, error) {
...
+			someIP[:12] + ":8333",
+			fmtErr,
+		},
+		{
+			someIP + ":abcd",
+			fmtErr,
+		},
+	}
+
+	amgr := addrmgr.New("", nil)
```
Comment (davecgh) [inline addrmgr/addrmanager_test.go]: Shouldn't this have a name so the created manager has a reasonable name?

**Comment by jimmysong** [comment]
@davecgh fixed as requested.

**Comment by davecgh** [comment]
Thanks.  OK.

**Pull request merged**